File: JavaPractice/src/FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.java
import java.util.HashMap;

public class FindPlayerScoreHighestToLowest_HighToLow_LowToHigh {

	public static class Players {
		
		private String name = null;
		private int score = 0;
		private int minscore = Integer.MAX_VALUE;
		private int maxscore = Integer.MIN_VALUE;
		
		public Players(String name, int score) {
			setName(name);
			setScore(name, score);
		}
		public String getName() {
			return name;
		}
		public int getScore() {
			return score;
		}
		public void setScore(String name, int score) {
			this.name = name;
			this.score = score;
			setScoreValue(name, score);
		}
		public void setName(String name) {
			this.name = name;
		}
		public int getMinscore() {
			return minscore;
		}
		private void setMinscore(int minscore) {
			//setScoreValue(getName(),minscore);
			this.minscore = minscore;
		}
		public int getMaxscore() {
			return maxscore;
		}
		private void setMaxscore(int maxscore) {
			//setScoreValue(getName(),maxscore);
			this.maxscore = maxscore;
		}
		public void setScoreValue(String name, int score) {
			
			setName(name);
			//setScore(name,score);
			if(score < getMinscore()) {
				setMinscore(score);
			}
			if(score > getMaxscore()) {
				setMaxscore(score);
			}
		}
		public void printHighToLow() {
			
		}
		public void printLowToHigh() {
			
		}
		
	}
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		HashMap<String, Players> hm = new HashMap<>();
		FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.Players pl = new FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.Players("Divyesh", 5);
		FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.Players p2 = new FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.Players("Jinal", 15);
		if(!hm.containsKey("Divyesh"))
			hm.put("Divyesh", pl);
		if(!hm.containsKey("Jinal"))
			hm.put("Jinal", p2);
		System.out.println(hm.get("Divyesh").getScore());
		System.out.println(hm.get("Divyesh").getMaxscore());
		System.out.println(hm.get("Divyesh").getMinscore());
		System.out.println(hm.get("Jinal").getScore());
		System.out.println(hm.get("Jinal").getMaxscore());
		System.out.println(hm.get("Jinal").getMinscore());

		
		
		
	}
}

File: PythonPractice/src/FindPlayerScoreHighestToLowest_HighToLow_LowToHigh.py
class Players:
    def __init__(self, name: str, score: int):
        if not name or score is None:
            raise ValueError("Invalid input: Name and score must be provided")
        self.name = name
        self.score = score
        self.minscore = score
        self.maxscore = score

    def get_name(self) -> str:
        return self.name

    def get_score(self) -> int:
        return self.score

    def set_score(self, name: str, score: int):
        if not name or score is None:
            raise ValueError("Invalid input: Name and score must be provided")
        self.name = name
        self.score = score
        self.set_score_value(score)

    def set_name(self, name: str):
        if not name:
            raise ValueError("Invalid input: Name must be provided")
        self.name = name

    def get_minscore(self) -> int:
        return self.minscore

    def set_minscore(self, minscore: int):
        self.minscore = minscore

    def get_maxscore(self) -> int:
        return self.maxscore

    def set_maxscore(self, maxscore: int):
        self.maxscore = maxscore

    def set_score_value(self, score: int):
        if score < self.get_minscore():
            self.set_minscore(score)
        if score > self.get_maxscore():
            self.set_maxscore(score)

    def print_high_to_low(self):
        pass

    def print_low_to_high(self):
        pass


def main():
    hm = {}
    pl = Players("Divyesh", 5)
    p2 = Players("Jinal", 15)

    if "Divyesh" not in hm:
        hm["Divyesh"] = pl
    if "Jinal" not in hm:
        hm["Jinal"] = p2

    print(hm["Divyesh"].get_score())
    print(hm["Divyesh"].get_maxscore())
    print(hm["Divyesh"].get_minscore())
    print(hm["Jinal"].get_score())
    print(hm["Jinal"].get_maxscore())
    print(hm["Jinal"].get_minscore())


if __name__ == "__main__":
    main()
